CMakeLists.txt

```diff
@@ -11,7 +11,8 @@ foreach(var ${vars})
     list(APPEND STAGE0_ARGS "-D${CMAKE_MATCH_1}=${${var}}")
   elseif("${currentHelpString}" MATCHES "No help, variable specified on the command line." OR "${currentHelpString}" STREQUAL "")
     list(APPEND CL_ARGS "-D${var}=${${var}}")
-    if("${var}" STREQUAL "USE_GMP")
+    set(FORWARDED_OPTIONS "USE_GMP" "LLVM_CONFIG" "LLVM")
+    if("${var}" IN_LIST FORWARDED_OPTIONS)
```

Comment:
This should be reverted again now, right? Not having to worry about linking LLVM twice for the cross builds is actually a good reason not to link LLVM in stage 0.

Reply:
@Kha when exactly does `FORWARDED_OPTIONS` kick in? Suppose we perform:
```
lean4/$ mkdir build && cd build && cmake ../ (1)
lean4/build$ make -j # (2)
```

Is it correct that:

- The `FORWARDED_OPTIONS` in `CmakeLists.txt` in configured in step `(1)`.
- The `src/CMakeLists.txt`,  runs next, which sets up the `stage1` build. 
- The `stage0/src/CMakelists.txt` runs at step `(2)`, which re-runs its private copy of `CMake`? It is at this step that `FORWARDED_OPTIONS` is used to forward the options from the "outer" `Cmake` in step `(1)` to the "inner" `Cmake` in step (2)?

Reply:
AFAIK all stages are configured in (2), as they are all set up using `ExternalProject_add`. But it doesn't really matter in practice when that happens.

Reply:
I'm trying to build a mental model of the build system's moving parts, and this is very helpful.  Did I get the logical flow of information correct, from `FORWARDED_OPTIONS` in `CMakeLists.txt` to `src/stage0/CMakeLists.txt`?

Reply:
Yes, that is how `ExternalProject` works